modules/core/src/weaver/suites.scala

```diff
@@ -114,6 +126,20 @@ abstract class MutableFSuite[F[_]] extends RunnableSuite[F]  {
       } yield result
     }
 
+  private[this] def onlyNotOnCiFailure(test: TestName): TestOutcome = {
+    val result = Result.Failure(
+      msg = "'Only' tag is not allowed when `isCI=true`",
+      source = None,
+      location = List(test.location)
+    )
+    TestOutcome(
+      name = test.name,
+      duration = FiniteDuration(0, "ns"),
+      result = result,
+      log = Chain.empty
+    )
+  }
```

Comment:
I think this makes sense for making a failing test outcome, but if there is a better way, please let me know.

A few things that I wanted to clarify
- msg in the result is the same message used in Munit
- source in result is None, this was't caused by an exception so no need to provide an exception (right?)
- location in result, using the location of the test definition itself as a single entry, is this what is expected here?
- duration is TestOutcome set to 0 since this test didn't actually run - is this fine?
- I provided an empty chain for TestOutcome.log, but do we want something here?